app/controllers/auth_controller.rb

```diff
@@ -20,8 +20,16 @@ def create
     when :google
       user = GoogleAuthService.find_user_by_token(data_attrs[:google_id_token], data_attrs[:create_user])
       AuthToken.new(user: user)
-    when :okta, :facebook
-      user = "::#{method.to_s.capitalize}AuthService".constantize.find_user_by_token(data_attrs[:"#{method}_access_token"], data_attrs[:create_user])
+    when :okta
+      user = OktaAuthService.find_user_by_token(data_attrs[:okta_access_token], data_attrs[:create_user])
+      AuthToken.new(user: user)
+    when :facebook
+      # Handle both access tokens and ID tokens for Facebook
+      if data_attrs[:facebook_access_token]
```

Comment:
[Must-fix] Missing validation when both facebook_access_token and facebook_id_token are provided. Evidence: Lines 28-32 allow both tokens to be processed. Impact: Could lead to AuthToken.new(user: nil) causing downstream issues. Fix: Add mutual exclusion validation. [Category: File]

Reply:
I would actually say a better way to refactor this is to get rid of the extraction of the credential handler, and just do a big switch based upon which credential is in the request. that way we don't need to have a special case for facebook when the already existing default fallback of code could handle it just fine.